dockstore-client/src/main/java/io/dockstore/client/cli/YamlClient.java

```diff
@@ -0,0 +1,102 @@
+import com.beust.jcommander.ParameterException;
+import com.beust.jcommander.Parameters;
+import io.dockstore.client.cli.YamlVerifyUtility.ValidateYamlException;
+import java.util.List;
+
+public final class YamlClient {
+
+    public static final String VALIDATE_HELP_MESSAGE = "Verifies " + DOCKSTOREYML + " is correct and ensures all required files are present";
+    public static final String ERROR_NO_COMMAND = "ERROR: No command given";
+
+
+
+    /**
+     * @param args
+     */
+    public boolean handleCommand(List<String> args) {
+        String[] argv = args.toArray(new String[args.size()]);
+        JCommander jc = new JCommander();
+
+        CommandYaml commandYaml = new CommandYaml();
+        JCommander jcPlugin = JCommanderUtility.addCommand(jc, YAML, commandYaml);
+        CommandYamlValidate commandYamlValidate = new CommandYamlValidate();
+        JCommanderUtility.addCommand(jcPlugin, YamlVerifyUtility.COMMAND_NAME, commandYamlValidate);
+        try {
+            jcPlugin.parse(argv);
+        } catch (ParameterException ex) {
+            if (YamlVerifyUtility.COMMAND_NAME.equals(ex.getJCommander().getParsedCommand())) {
+                printJCommanderHelp(jcPlugin, "dockstore " + YAML, YamlVerifyUtility.COMMAND_NAME);
+                out(ex.getMessage());
+            } else {
+                printJCommanderHelp(jc, "dockstore", YAML);
+                out(ex.getMessage());
+            }
+            return true;
+        }
+        if (commandYaml.help) {
+            printJCommanderHelp(jc, "dockstore", YAML);
+        } else if  (jcPlugin.getParsedCommand() == null) {
+            printJCommanderHelp(jc, "dockstore", YAML);
+            out(ERROR_NO_COMMAND);
+        } else {
+            switch (jcPlugin.getParsedCommand()) {
+            case YamlVerifyUtility.COMMAND_NAME:
+                if (commandYamlValidate.help) {
+                    printJCommanderHelp(jcPlugin, "dockstore " + YAML, YamlVerifyUtility.COMMAND_NAME);
+                } else {
+                    try {
+                        YamlVerifyUtility.dockstoreValidate(commandYamlValidate.path);
+                    } catch (ValidateYamlException ex) {
+                        out(ex.getMessage());
+                    }
+                }
+                break;
+            default:
+                // fall through
+            }
+
+        }
+        return true;
+    }
+
+
+    @Parameters(separators = "=", commandDescription = "Tools used for " + YAML + " files")
+    private class CommandYaml {
+        @Parameter(names = "--help", description = "Prints help for " + YAML + " command", help = true)
+        private boolean help = false;
+    }
+
+    @Parameters(separators = "=", commandDescription = VALIDATE_HELP_MESSAGE)
+    private class CommandYamlValidate {
+        @Parameter(names = "--path", description = "Directory that contains " + DOCKSTOREYML + " (ex. /home/usr/Dockstore/test, ~/Dockstore/test, or ../test)", required = true)
```

Comment:
Really nitpicky and ignorable, especially since I've beat this one to death :) 

1. Pick a more informative name than `test`, e.g., `my-workflow-repo`, `workflow-repo`, `biotool`, etc. (you're not running this on a random .dockstore.yml somewhere on your system, but in the context of a .dockstore.yml that's part of tree that is or is going to be in GitHub.
2. Give a .github example.

Reply:
Changed it so the help message is now,
```
Required parameters:
  --path <path>  Directory that contains .dockstore.yml (ex. /home/usr/workflow-repo, ~/Dockstore/.github, or ../biotool)
```